pkg/controller/storagecluster/reconcile.go

```diff
@@ -301,6 +361,22 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 	return nil
 }
 
+// determineFailureDomain determines the appropriate Ceph failure domain based
+// on the storage cluster's topology map
+func determineFailureDomain(topologyMap *ocsv1.NodeTopologyMap) string {
+	failureDomain := "host"
+
+	for label, labelValues := range topologyMap.Labels {
+		if strings.Contains(label, "zone") {
+			if len(labelValues) >= 3 {
```

Comment:
At some point it would be great to replace the hard-coded `3` here with the replica number from the storageDeviceSet.